File: generator/src/test/resources/swift/expected/RamlTypeAnnotationsTest/test-patchable-class-generation.output.swift
import Sunday

public class Test : Codable, CustomDebugStringConvertible {

  public var string: UpdateOp<String>?
  public var int: UpdateOp<Int>?
  public var bool: UpdateOp<Bool>?
  public var nullable: PatchOp<String>?
  public var optional: UpdateOp<String>?
  public var nullableOptional: PatchOp<String>?
  public var debugDescription: String {
    return DescriptionBuilder(Test.self)
        .add(string, named: "string")
        .add(int, named: "int")
        .add(bool, named: "bool")
        .add(nullable, named: "nullable")
        .add(optional, named: "optional")
        .add(nullableOptional, named: "nullableOptional")
        .build()
  }

  public init(
    string: UpdateOp<String>? = .none,
    int: UpdateOp<Int>? = .none,
    bool: UpdateOp<Bool>? = .none,
    nullable: PatchOp<String>? = .none,
    optional: UpdateOp<String>? = .none,
    nullableOptional: PatchOp<String>? = .none
  ) {
    self.string = string
    self.int = int
    self.bool = bool
    self.nullable = nullable
    self.optional = optional
    self.nullableOptional = nullableOptional
  }

  public required init(from decoder: Decoder) throws {
    let container = try decoder.container(keyedBy: CodingKeys.self)
    self.string = try container.decodeIfExists(String.self, forKey: .string)
    self.int = try container.decodeIfExists(Int.self, forKey: .int)
    self.bool = try container.decodeIfExists(Bool.self, forKey: .bool)
    self.nullable = try container.decodeIfExists(String.self, forKey: .nullable)
    self.optional = try container.decodeIfExists(String.self, forKey: .optional)
    self.nullableOptional = try container.decodeIfExists(String.self, forKey: .nullableOptional)
  }

  public func encode(to encoder: Encoder) throws {
    var container = encoder.container(keyedBy: CodingKeys.self)
    try container.encodeIfExists(self.string, forKey: .string)
    try container.encodeIfExists(self.int, forKey: .int)
    try container.encodeIfExists(self.bool, forKey: .bool)
    try container.encodeIfExists(self.nullable, forKey: .nullable)
    try container.encodeIfExists(self.optional, forKey: .optional)
    try container.encodeIfExists(self.nullableOptional, forKey: .nullableOptional)
  }

  public func withString(string: UpdateOp<String>?) -> Test {
    return Test(string: string, int: int, bool: bool, nullable: nullable, optional: optional,
        nullableOptional: nullableOptional)
  }

  public func withInt(int: UpdateOp<Int>?) -> Test {
    return Test(string: string, int: int, bool: bool, nullable: nullable, optional: optional,
        nullableOptional: nullableOptional)
  }

  public func withBool(bool: UpdateOp<Bool>?) -> Test {
    return Test(string: string, int: int, bool: bool, nullable: nullable, optional: optional,
        nullableOptional: nullableOptional)
  }

  public func withNullable(nullable: PatchOp<String>?) -> Test {
    return Test(string: string, int: int, bool: bool, nullable: nullable, optional: optional,
        nullableOptional: nullableOptional)
  }

  public func withOptional(optional: UpdateOp<String>?) -> Test {
    return Test(string: string, int: int, bool: bool, nullable: nullable, optional: optional,
        nullableOptional: nullableOptional)
  }

  public func withNullableOptional(nullableOptional: PatchOp<String>?) -> Test {
    return Test(string: string, int: int, bool: bool, nullable: nullable, optional: optional,
        nullableOptional: nullableOptional)
  }

  fileprivate enum CodingKeys : String, CodingKey {

    case string = "string"
    case int = "int"
    case bool = "bool"
    case nullable = "nullable"
    case optional = "optional"
    case nullableOptional = "nullableOptional"

  }

}

extension AnyPatchOp where Value == Test {

  public static func merge(
    string: UpdateOp<String>? = .none,
    int: UpdateOp<Int>? = .none,
    bool: UpdateOp<Bool>? = .none,
    nullable: PatchOp<String>? = .none,
    optional: UpdateOp<String>? = .none,
    nullableOptional: PatchOp<String>? = .none
  ) -> Self {
    Self.merge(Test(string: string, int: int, bool: bool, nullable: nullable, optional: optional,
        nullableOptional: nullableOptional))
  }

}

File: generator/src/test/resources/kotlin/expected/RamlTypeAnnotationsTest/test-patchable-class-generation.output.kt
package io.test

import com.fasterxml.jackson.`annotation`.JsonInclude
import io.outfoxx.sunday.json.patch.Patch
import io.outfoxx.sunday.json.patch.PatchOp
import io.outfoxx.sunday.json.patch.UpdateOp
import kotlin.Any
import kotlin.Boolean
import kotlin.Int
import kotlin.String
import kotlin.Unit

@JsonInclude(JsonInclude.Include.NON_EMPTY)
public open class Test(
  public var string: UpdateOp<String> = PatchOp.none(),
  public var int: UpdateOp<Int> = PatchOp.none(),
  public var bool: UpdateOp<Boolean> = PatchOp.none(),
  public var nullable: PatchOp<String> = PatchOp.none(),
  public var optional: UpdateOp<String> = PatchOp.none(),
  public var nullableOptional: PatchOp<String> = PatchOp.none(),
) : Patch {
  override fun hashCode(): Int {
    var result = 1
    result = 31 * result + string.hashCode()
    result = 31 * result + int.hashCode()
    result = 31 * result + bool.hashCode()
    result = 31 * result + nullable.hashCode()
    result = 31 * result + optional.hashCode()
    result = 31 * result + nullableOptional.hashCode()
    return result
  }

  override fun equals(other: Any?): Boolean {
    if (this === other) return true
    if (javaClass != other?.javaClass) return false

    other as Test

    if (string != other.string) return false
    if (int != other.int) return false
    if (bool != other.bool) return false
    if (nullable != other.nullable) return false
    if (optional != other.optional) return false
    if (nullableOptional != other.nullableOptional) return false

    return true
  }

  override fun toString(): String = """
  |Test(string='$string',
  | int='$int',
  | bool='$bool',
  | nullable='$nullable',
  | optional='$optional',
  | nullableOptional='$nullableOptional')
  """.trimMargin()

  public companion object {
    public inline fun merge(`init`: Test.() -> Unit): PatchOp.Set<Test> {
      val patch = Test()
      patch.init()
      return PatchOp.Set(patch)
    }

    public inline fun patch(`init`: Test.() -> Unit): Test = merge(init).value
  }
}
